numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -2047,23 +2047,19 @@ array_empty_like(PyObject *NPY_UNUSED(ignored),
         goto fail;
     }
     /* steals the reference to dt_info.descr if it's not NULL */
+    if (dt_info.descr != NULL) {
+        Py_INCREF(dt_info.descr);
+    }
     ret = (PyArrayObject *)PyArray_NewLikeArrayWithShape(
             prototype, order, dt_info.descr, dt_info.dtype,
             shape.len, shape.ptr, subok);
     npy_free_cache_dim_obj(shape);
-    if (!ret) {
```

Comment:
Is `ret` known not to be `NULL`.

Reply:
Νο, it can be NULL. This `if` however only serves the purpose of skipping refcounting for `dt_info.desc`, which we don't want now.